Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -6669,6 +6669,30 @@ RefPtr<CSSValue> consumeScrollSnapType(CSSParserTokenRange& range)
     return CSSValueList::createSpaceSeparated(firstValue.releaseNonNull());
 }
 
+RefPtr<CSSValue> consumeScrollbarGutter(CSSParserTokenRange& range)
+{
+    auto firstValue = consumeIdent<CSSValueAuto, CSSValueStable, CSSValueBothEdges>(range);
+    if (!firstValue)
+        return nullptr;
+
+    auto secondValue = consumeIdent<CSSValueBothEdges, CSSValueStable>(range);
+
+    if (!secondValue && firstValue->valueID() == CSSValueBothEdges)
+        return nullptr;
+
+    if (!secondValue)
+        return CSSValueList::createSpaceSeparated(firstValue.releaseNonNull());
+
+    if (firstValue->valueID() == CSSValueStable && secondValue->valueID() == CSSValueBothEdges)
+        return CSSValueList::createSpaceSeparated(firstValue.releaseNonNull(), secondValue.releaseNonNull());
+
+    if (firstValue->valueID() == CSSValueBothEdges && secondValue->valueID() == CSSValueStable)
+        return CSSValueList::createSpaceSeparated(secondValue.releaseNonNull(), firstValue.releaseNonNull());
+
+    // Only valid cobination of two values is stable && both-edges or both-edges && stable
+    return nullptr;
```

Comment:
I would have written it like this:

```suggestion
    if (auto ident = consumeIdent<CSSValueAuto>(range))
        return CSSPrimitiveValue::create(CSSValueAuto);
    
    if (auto first = consumeIdent<CSSValueStable>(range)) {
        if (auto second = consumeIdent<CSSValueBothEdges>(range))
            return CSSValuePair::create(...);
        return first;
    }
 
    if (auto first = consumeIdent<CSSValueBothEdges>(range)) {
        if (auto second = consumeIdent<CSSValueStable>(range))
            return CSSValuePair::create(...);
        return nullptr;
    }
    
    return nullptr;
```